subprojects/analysis-markdown-jb/src/main/kotlin/org/jetbrains/dokka/analysis/markdown/jb/MarkdownParser.kt

```diff
@@ -77,6 +77,30 @@ public open class MarkdownParser(
             ).flatMap { it.children }
         )
 
+    /**
+     * Handler for [MarkdownTokenTypes.ATX_CONTENT], which is the content of the header
+     * elements like [MarkdownElementTypes.ATX_1], [MarkdownElementTypes.ATX_2] and so on.
+     *
+     * For example, a header line like `# Header text` is expected to be parsed into:
+     * - One [MarkdownTokenTypes.ATX_HEADER] with startOffset = 0, endOffset = 1 (only the `#` symbol)
+     * - Composite [MarkdownTokenTypes.ATX_CONTENT] with four children: WHITE_SPACE, TEXT, WHITE_SPACE, TEXT.
+     */
+    private fun headerContentHandler(node: ASTNode): List<DocTag> {
+        // ATX_CONTENT contains everything after the `#` symbol, so if there's a space
+        // in-between the `#` symbol and the text (like `# header`), it will be present here too.
+        // However, we don't need the first space between the `#` symbol and the text,
+        // so we just skip it (otherwise the header text will be parsed as `<whitespace>header` instead of `header`).
+        // If there's more space between `#` and text, like `#     header`, it will still be a single WHITE_SPACE
+        // element, but it will be wider, so the solution below should still hold.
+        val textStartsWithWhitespace = node.children.firstOrNull()?.type == MarkdownTokenTypes.WHITE_SPACE
```

Comment:
It should trim not only the beginning.
> Leading and trailing spaces or tabs are ignored in parsing inline content:

See https://spec.commonmark.org/0.30/#atx-headings

Reply:
Good thought, fixed